test/kubernetes/integration_test.go

```diff
@@ -358,6 +358,22 @@ func TestSnapshot_Create(t *testing.T) {
 			Name: "my-csi-app-2",
 		},
 		Spec: v1.PodSpec{
+			InitContainers: []v1.Container{
+				{
+					Name:  "my-csi",
+					Image: "busybox",
+					VolumeMounts: []v1.VolumeMount{
+						{
+							MountPath: "/data",
+							Name:      volumeName,
+						},
+					},
+					Command: []string{
+						"sh", "-c",
+						"echo testcanary > /data/canary",
+					},
```

Comment:
Why do we have these init containers? I couldn't figure out why we need them.

Reply:
Added a comment, but basically, waitForPod will consider it successful once the pod enters the Running state. Regular containers enter the running state once the healthcheck passes, so by using an initcontainer, the pod will never enter the running state if the assertion fails.